native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java

```diff
@@ -111,16 +111,17 @@ public static Object getNext(BObject channel) {
         }
     }
 
-    public static Object readRecord(BObject channel, BString separator) {
-        BufferedReader bufferedReader = (BufferedReader)
-                channel.getNativeData(BUFFERED_READER_ENTRY);
+    public static Object readRecord(BObject channel) {
+        BufferedReader bufferedReader = (BufferedReader) channel.getNativeData(BUFFERED_READER_ENTRY);
         try {
             String line = bufferedReader.readLine();
             if (line == null) {
                 bufferedReader.close();
                 return IOUtils.createEoFError();
             }
-            String[] record = line.strip().split(separator.getValue());
+            DelimitedRecordChannel textRecordChannel =
+                    (DelimitedRecordChannel) channel.getNativeData(TXT_RECORD_CHANNEL_NAME);
+            String[] record = textRecordChannel.getFields(line);
```

Comment:
`DelimitedRecordChannel.getFields()` is required because simply using `split()` does not account for situations,
- where there are delimiters inside quoted fields
- the delimiter is not a comma, and hence has to be derived from the `DelimitedRecordChannel`